modules/transport-netty4/src/main/java/org/elasticsearch/transport/netty4/Netty4TcpChannel.java

```diff
@@ -88,7 +92,25 @@ public static ChannelPromise addPromise(ActionListener<Void> listener, Channel c
 
     @Override
     public void close() {
-        channel.close();
+        if (rstOnClose) {
+            rstAndClose();
+        } else {
+            channel.close();
+        }
+    }
+
+    private void rstAndClose() {
+        Releasables.close(() -> {
+            if (channel.isOpen()) {
+                try {
+                    channel.config().setOption(ChannelOption.SO_LINGER, 0);
```

Comment:
I believe this brings back the logging mentioned in https://github.com/elastic/elasticsearch/issues/82360 - I think we need to drop `Connection reset` logs to `DEBUG` if `transport.netty.rst_on_close` is set.

Reply:
Ah right It does ... looking into a way that adjusts the level (slightly tricky to find a nice way) ...

Reply:
Alright worked around this now by adding a setting for all transports (though only Netty does pick it up). Not super happy with the approach but it works. We could also go with just a static boolean via a property but I tend to rather make things work out of the box in my idea :D